packages/gatsby/src/utils/webpack.config.js

```diff
@@ -451,47 +451,61 @@ module.exports = async (program, directory, suppliedStage) => {
   }
 
   if (stage === `build-html` || stage === `develop-html`) {
+    // Packages we want to externalize to save space
     const externalList = [
-      // match `lodash` and `lodash/foo`
-      // but not things like `lodash-es`
+      `common-tags`,
+      /^core-js\//,
+      `crypto`,
+      `debug`,
+      `fs`,
+      `https`,
+      `http`,
       `lodash`,
+      `path`,
+      `semver`,
       /^lodash\//,
+      `zlib`,
+    ]
+
+    // Packages we want to externalize because the user should provide them
```

Comment:
This is mix of should and could:
- `react` and `react-dom` is expected to be provided by user (marked as gatsby's peerDependencies) - example of "should"
- `react-helmet` is not dep or peerDep of gatsby, but it's used in majority of gatsby projects to handle <head> tags - example of "could"